frontend/src/components/NavDropdown.tsx

```diff
@@ -31,37 +22,61 @@ export default function NavDropdown({
   menuItems,
   setActiveNavDropdownIndex,
 }: NavDropdownProps): JSX.Element {
-  let isOpen = activeNavDropdownIndex === index;
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+
+  function eventHandler(e: any) {
+    const dropdowns = document.getElementsByName("navDropDownButton");
+    let dropdownClicked = false;
+    for (let dropdown of dropdowns) {
+      if (dropdown.contains(e.target)) {
+        dropdownClicked = true;
+      }
+    }
+
+    let targetId = null;
+    if (dropdownClicked) {
+      let targetNode = e.target;
+      if (e.target.localName === "span") {
+        targetNode = targetNode.parentNode;
+      }
+      if (!targetNode.className.includes("simpler-subnav-open")) {
+        targetId = toNumber(targetNode.id);
+      }
+    }
+    setActiveNavDropdownIndex(targetId);
+  }
 
-  function handleToggle() {
-    const activeIndex: IndexType = isOpen ? null : index;
+  function handleToggle(e: any) {
+    let activeIndex: IndexType = isOpen ? null : index;
+    if (!!activeIndex) {
+      e.stopPropagation();
+      requestAnimationFrame(() =>
+        document.addEventListener(
+          "click",
+          function (e) {
+            eventHandler(e);
+          },
+          { once: true },
+        ),
+      );
+    }
     setActiveNavDropdownIndex(activeIndex);
   }
 
   useEffect(() => {
-    isOpen = activeNavDropdownIndex === index;
+    setIsOpen(activeNavDropdownIndex === index);
   }, [activeNavDropdownIndex, index]);
 
   return (
     <>
       <NavDropDownButton
+        id={index.toString()}
```

Comment:
I see what you're doing, and it's not a bad solution, but since ids need to be unique in the document I'm not a big fan of using just a blank number for an id. Can we set this as `nav-dropdown-button-${index}` and then parse the index out of that in the handler?

Reply:
done!